src/document.c

```diff
@@ -439,20 +452,40 @@ FIELD_PREPROCESSOR(fulltextPreprocessor) {
     if (FieldSpec_IsPhonetics(fs)) {
       options |= TOKENIZE_PHONETICS;
     }
-    aCtx->tokenizer->Start(aCtx->tokenizer, (char *)c, fl, options);
 
-    Token tok = {0};
-    uint32_t newTokPos;
-    while (0 != (newTokPos = aCtx->tokenizer->Next(aCtx->tokenizer, &tok))) {
-      forwardIndexTokenFunc(&tokCtx, &tok);
+    unsigned int multiTextOffsetDelta;
+    if (valueCount > 1 && RSGlobalConfig.multiTextOffsetDelta > 0) {
+      multiTextOffsetDelta = RSGlobalConfig.multiTextOffsetDelta - 1;
+    } else {
+      multiTextOffsetDelta = 0;
     }
-    uint32_t lastTokPos = aCtx->tokenizer->ctx.lastOffset;
 
-    if (curOffsetField) {
-      curOffsetField->lastTokPos = lastTokPos;
+    for (size_t i = 0; i < valueCount; ++i) {
+    
+      // Already got the first value
+      if (i) {        
+        c = DocumentField_GetArrayValueCStr(field, &fl, i);
```

Comment:
Where did you got the first value?

Reply:
Some line above
`const char *c = DocumentField_GetValueCStr(field, &fl);`

Reply:
So `DocumentField_GetValueCStr` will return the first field if its an array?

Reply:
Yes
```
/* return value as c string (if array - return the first entry)*/
const char *DocumentField_GetValueCStr(const DocumentField *df, size_t *len);
```